equinix/resource_metal_connection.go

```diff
@@ -383,6 +383,11 @@ func resourceMetalConnectionRead(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
+	vlans := getConnectionVlans(conn)
```

Comment:
nit: you can set `"vlans":              getConnectionVlans(conn),` directly in set map without check in for nil

Reply:
I started with adding the attribute in `setMap`, but including the method call directly in `setMap` sets the `vlans` value to `[]` instead of leaving the attribute unset.  In that case, a `metal_connection` that has no `vlans` specified will always show changes in `terraform plan` b/c terraform wants to unset `[]`.

It's possible we could get around that by specifying a default value of `[]` for `vlans`?  I'm not sure if changing/adding that default would cause problems elsewhere, so I went with the `nil` check to minimize impact on existing code.